addons/udes_stock/models/stock_move.py

```diff
@@ -519,3 +519,53 @@ def refactor_by_move_line_groups(self, groups):
             })
 
         return result_moves
+
+    @api.multi
+    def refactor_action_batch_pickings_by_date_priority(self):
+        """Batch pickings by date and priority."""
+        self._refactor_action_batch_pickings_by(
+            lambda picking: (picking.scheduled_date, picking.priority)
+        )
+
+    @api.multi
+    def refactor_action_batch_pickings_by_date(self):
+        """Batch pickings by date."""
+        self._refactor_action_batch_pickings_by(
+            lambda picking: (picking.scheduled_date,)
+        )
+
+    def _refactor_action_batch_pickings_by(self, by_key):
+        """Group picks in batches.
+
+        Args:
+            by_key (function): The function to generate the key to group pickings by.
+                Should return a :obj:`tuple`.
+        """
+        PickingBatch = self.env["stock.picking.batch"]
+
+        # Precache pickings.
+        self.mapped("picking_id")
```

Comment:
Not sure this is needed, since we are doing the same in line 562 and we dont't have any other access to the pickings of the moves until that.